lib/storage/partition.go

```diff
@@ -1796,10 +1796,16 @@ func appendPartsToMerge(dst, src []*partWrapper, maxPartsToMerge int, maxOutByte
 	// Exhaustive search for parts giving the lowest write amplification when merged.
 	var pws []*partWrapper
 	maxM := float64(0)
+	outSize := uint64(0)
 	for i := minSrcParts; i <= maxSrcParts; i++ {
+		outSize = uint64(0)
 		for j := 0; j <= len(src)-i; j++ {
 			a := src[j : j+i]
-			outSize := getPartsSize(a)
+			if outSize == 0 {
+				outSize = getPartsSize(a)
+			} else {
+				outSize += src[j+i-1].p.size - src[j-1].p.size
+			}
```

Comment:
The updated logic is much harder to track and understand than the original one. This cost can be justified only if it brings significant performance improvements. It would be great to add benchmarks, which clearly show these improvements then.

Reply:
OK, I added benchmarks but it didn't look like a obviously improvement.

code:
```go
func BenchmarkAppendPartsToMerge(b *testing.B) {
	var sizes []uint64
	maxOutSize := uint64(0)
	r := rand.New(rand.NewSource(1))
	for i := 0; i < 1024; i++ {
		n := uint64(uint32(r.NormFloat64() * 1e9))
		if n < 0 {
			n = -n
		}
		n++
		maxOutSize += n
		sizes = append(sizes, n)
	}
	pwsOld := newTestPartWrappersForSizes(sizes)
	pwsNew := newTestPartWrappersForSizes(sizes)

	// Annotate the sort operation in the appendPartsToMerge method，and move here
	sortPartsForOptimalMerge(pwsOld)
	sortPartsForOptimalMerge(pwsNew)

	b.Run("old", func(b *testing.B) {
		for i := 0; i < b.N; i++ {
			appendPartsToMerge(nil, pwsOld, defaultPartsToMerge, maxOutSize)
		}
	})
	b.Run("new", func(b *testing.B) {
		for i := 0; i < b.N; i++ {
			appendPartsToMerge(nil, pwsNew, defaultPartsToMerge, maxOutSize)
		}
	})

}
```

result:
```
goos: darwin
goarch: amd64
pkg: github.com/VictoriaMetrics/VictoriaMetrics/lib/storage
cpu: Intel(R) Core(TM) i7-1060NG7 CPU @ 1.20GHz
BenchmarkName
BenchmarkName/old
BenchmarkName/old-8         	    8438	    123996 ns/op
BenchmarkName/new
BenchmarkName/new-8         	   10000	    109111 ns/op
PASS
```